apps/flame-defi/app/features/cosmos-wallet/contexts/cosmos-wallet-context.tsx

```diff
@@ -114,6 +122,26 @@ export const CosmosWalletProvider: React.FC<CosmosWalletProviderProps> = ({
   const { balance: cosmosBalance, isLoading: isLoadingCosmosBalance } =
     useBalancePolling(getBalanceCallback, pollingConfig);
 
+  useEffect(() => {
+    if (cosmosBalance) {
+      const usdcToken = currencies?.find(
+        (currency) => currency.coinDenom.toLowerCase() === "usdc",
+      );
+      getQuote(
+        TRADE_TYPE.EXACT_IN,
+        { token: currencies[0], value: removeNonNumeric(cosmosBalance.value) },
```

Comment:
Stripping decimals instead of scaling the value based on the defined decimals seems like a good place for a bug to hang out in. Is this the right approach?

Reply:
This is not stripping decimals. The cosmosBalance.value is turned like this "1.23 TIA". We have to remove the TIA and only pass the number value

Reply:
Oh man, I'm tired. I'm sorry! I misread the regex hard.